src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -126,11 +131,52 @@ std::shared_ptr<uint32_t> GetCounterHolder(const char * name)
     return std::make_shared<uint32_t>(0);
 }
 
+bool IsSRPType(const char * domain)
+{
+    return strcmp(kSRPDot, domain) == 0;
+}
+
 } // namespace
 
 namespace chip {
 namespace Dnssd {
 
+/**
+ * @brief Callback that is called when the timeout for resolving on the kSRPDot domain has expired.
+ *
+ * @param[in] systemLayer The system layer.
+ * @param[in] callbackContext The context passed to the timer callback.
+ */
+void SRPTimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
+{
+    auto sdCtx = static_cast<ResolveContext *>(callbackContext);
+    VerifyOrDie(sdCtx != nullptr);
+    sdCtx->Finalize();
+}
+
+/**
+ * @brief Starts a timer to wait for the resolution on the kSRPDot domain to happen.
+ *
+ * @param[in] timeoutSeconds The timeout in seconds.
+ * @param[in] ResolveContext The resolve context.
+ */
+CHIP_ERROR StartSRPTimer(uint16_t timeoutInMSecs, ResolveContext * ctx)
+{
+    VerifyOrReturnValue(ctx != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    return DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds16(timeoutInMSecs), SRPTimerExpiredCallback,
+                                                 reinterpret_cast<void *>(ctx));
```

Comment:
```suggestion
                                                 static_cast<void *>(ctx));
```
or change the cast the other way to be `reinterpret_cast`, but make the two match.  I'd personally prefer `static_cast` for both.